catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/util/UiTestUtils.java

```diff
@@ -203,4 +206,28 @@ public static boolean comparePixelRgbaArrays(byte[] firstArray, byte[] secondArr
 		}
 		return true;
 	}
+
+	public static void waitForVariableValueEquals(UserVariable userVariable, double expectedValue, int timeout) {
+		double epsilon = 0.001;
+		int step = 10;
+		for (; timeout > 0; timeout -= step) {
+			if (Math.abs((double) userVariable.getValue() - expectedValue) < epsilon) {
+				return;
+			}
+			onView(isRoot()).perform(CustomActions.wait(step));
+		}
+		Assert.assertEquals(expectedValue, (double) userVariable.getValue(), epsilon);
+	}
+
+	public static void waitForVariableValueGreaterThan(UserVariable userVariable, double expectedValue, int timeout) {
+		double epsilon = 0.001;
+		int step = 10;
+		for (; timeout > 0; timeout -= step) {
+			if ((double) userVariable.getValue() > (expectedValue + epsilon)) {
```

Comment:
I don't think so.
With greater than we want the two values to be at least epsilon (plus a value >0) apart.
Saying value > expectedValue-epsilon would have the effect that value could be smaller than expectedValue.